### DISCUSSION TITLE
Questions related to the AMII APP Today I updated the IDE and before that there was a trick that when I closed these sounds were played to me

### DISCUSSION THREAD

**Discussion by AlekcWins** [ani-memes/AMII#134] [Q&A] [answered]
1)Today I updated the IDE and before that there was a trick that when I closed these sounds were played to me
-ara_ara_sayonara
-sayonara_bye_bye
-sayonara_senpai
**Question**: Does AMII have such this function?

2)Write in  Git that assets are now stored along the path:for 

1. wm -%appdata%\JetBrains\waifuMotivationAssets
2. AMII- %appdata%\JetBrains\amiiAssets

3) Why did you decide not to use JSON settings for sounds, but use folder path splices?

4)Judging by the code, there are about as many categories:

- ACKNOWLEDGEMENT(0),
-   FRUSTRATION(1),
-   ENRAGED(2),
-   CELEBRATION(3),
-   HAPPY(4),
-   SMUG(5),
-   WAITING(6),
-   MOTIVATION(7),
-   WELCOMING(8),
-   DEPARTURE(9),
-   ENCOURAGEMENT(10),
-   MOCKING(11),
-   SHOCKED(12),
-   DISAPPOINTMENT(13),
-   ALERT(14),
-   BORED(15),
-   TIRED(16),
-   PATIENTLY_WAITING(17),
-   POUTING(18),

**Question**: why are there a lot of unnecessary folders in the directory?
For example: 

- boried
- bored
-welcome
-welocome

**Comment by Unthrottled** [comment, marked as answer]
> Today I updated the IDE and before that there was a trick that when I closed these sounds were played to me
-ara_ara_sayonara
-sayonara_bye_bye
-sayonara_senpai
Question: Does AMII have such this function?

No, it currently does not.

>Write in Git that assets are now stored along the path:for
wm -%appdata%\JetBrains\waifuMotivationAssets
AMII- %appdata%\JetBrains\amiiAssets
Why did you decide not to use JSON settings for sounds, but use folder path splices?

Not 100% sure what you are asking for here. 
Visual assets may or may not reference an audio asset defined by the API. 
You can take a peek at the API design spec here https://github.com/ani-memes/amii-asset-management/blob/main/DesignSpec.md#api

Sample 

```
{"id":"9a8d07c228d234bd97227e6f65d8ffd1","path":"mocking/aqua_mocking_laugh.gif","alt":"","cat":[11],"char":["1eqq4979c"],"aud":"e30f629ebbf18655f6e8a7679f4d588c"}
```

```
{"id":"e30f629ebbf18655f6e8a7679f4d588c","path":"mockingaqua_mocking_laugh.wav"}
```

It appears that there may be a bug in the upload process in the asset management tool, I'll fix it, it's not the end of the world.


> Question: why are there a lot of unnecessary folders in the directory?

Because it doesn't matter what folder I put it in, I just need to define a unique key such that when it gets uploaded to the S3 bucket, that it won't overrwrite any existing asset. I use the web ui tool to manage all of the assets.

![image](https://user-images.githubusercontent.com/15972415/140653408-c956b871-7066-47bf-9444-ed7f8ec77931.png)
